### PR TITLE
Fix bounds check in import scanner

### PR DESCRIPTION
**Pull request by max-hoffman**
fixes: https://github.com/dolthub/dolt/issues/8495

The before code checks whether `s.i` is valid after doing the buffer access:
```go
	for ; unicode.IsSpace(rune(s.buf[s.i])); s.i++ {
		if s.i >= s.fill {
			s.read()
		}
	}
```
The after code inverts the check to make sure `s.i` is valid before indexing into the buffer.
```go
	for {
		if s.i >= s.fill {
			s.read()
		}
		if !unicode.IsSpace(rune(s.buf[s.i])) {
			break
		}
		s.i++
        }
```

### CODE REVIEW

**Comment by coffeegoddd** [comment]
@max-hoffman __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>77c38c3</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>77c38c3</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

File: go/cmd/dolt/commands/sql_statement_scanner_test.go [hunk 1]
```diff
@@ -187,6 +187,13 @@ insert into foo values (1,2,3)|`,
 			},
 			lineNums: []int{1, 2},
 		},
+		{
+			input: strings.Repeat(" ", 4096) + `insert into foo values (1,2,3)`,
```
Comment (zachmu) [inline go/cmd/dolt/commands/sql_statement_scanner_test.go:191]: Might put a link to the issue above this test in a commetn

**Review by zachmu — APPROVED** [review]
LGTM

**Comment by coffeegoddd** [comment]
@max-hoffman __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>dddfd9a</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>dddfd9a</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Pull request merged**